src/components/new-user-form/index.tsx

```diff
@@ -0,0 +1,211 @@
+import { ApolloError } from "@apollo/client";
+import React, { FormEvent, useState } from "react";
+import { formatPhone } from "../../utils/phone-formatter";
+import { validateBirthDate } from "../../validators/birthdate";
+import { validateConfirmPassword } from "../../validators/confirm-password";
+import { validateEmail } from "../../validators/email";
+import { validateName } from "../../validators/name";
+import { validatePassword } from "../../validators/password";
+import { validatePhone } from "../../validators/phone";
+import { Button } from "../button";
+import { Input } from "../input";
+import * as S from "./style";
+
+interface NewUserParamsTypes {
+  variables: {
+    data: {
+      name: string;
+      email: string;
+      phone: string;
+      birthDate: string;
+      password: string;
+      role: string;
+    };
+  };
+}
+
+interface NewUserFormTypes {
+  createUserFunction: (data: NewUserParamsTypes) => void;
+  loading: boolean;
+  error: ApolloError;
+}
+
+export const NewUserForm: React.FC<NewUserFormTypes> = ({
+  createUserFunction,
+  loading,
+  error,
+}) => {
+  const [name, setName] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const [phone, setPhone] = useState<string>("");
+  const [birthDate, setBirthDate] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState<string>("");
+  const [role, setRole] = useState<string>("");
+  const [nameError, setNameError] = useState<string>("");
+  const [emailError, setEmailError] = useState<string>("");
+  const [phoneError, setPhoneError] = useState<string>("");
+  const [birthDateError, setBirthDateError] = useState<string>("");
+  const [passwordError, setPasswordError] = useState<string>("");
+  const [passwordConfirmationError, setPasswordConfirmationError] =
+    useState<string>("");
```

Comment:
ao invés de vários states, o que acha de juntar os estados que são do `NewUserFormData` em um só?
```suggestion
  const [formData, setFormData] = useState<NewUserFormData>();
  const [role, setRole] = useState<string>("");
  const [nameError, setNameError] = useState<string>("");
  const [emailError, setEmailError] = useState<string>("");
  const [phoneError, setPhoneError] = useState<string>("");
  const [birthDateError, setBirthDateError] = useState<string>("");
  const [passwordError, setPasswordError] = useState<string>("");
  const [passwordConfirmationError, setPasswordConfirmationError] =
    useState<string>("");
    
 // pra atualizar o name, por exemplo
 setFormData((prevState) => ({ ...prevState, name: "Novo nome" }));
```

Reply:
Concordo demais, essa quantidade de states me incomoda um pouco, e pensando ainda em um form o caso de uso encaixa perfeitamente com um objeto em apenas um state. Obrigado Beto!